test/e2e-common.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests for the kn client.
+
+# If you already have the `KO_DOCKER_REPO` environment variable set and a
+# cluster setup and currently selected in your kubeconfig, call the script
+# with the `--run-tests` argument and it will use the cluster and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start Knative serving, run the tests and delete
+# the cluster.
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+DEFAULT_BUILD=latest
+
+# This is the variable saving the link to the YAML file to create and delete knative serving.
+KNATIVE_SERVING_RELEASE_YAML=""
+
+# Helper functions.
+
+function knative_setup() {
+  start_release_knative_serving "$KNATIVE_SERVING_VERSION"
```

Comment:
This function knative_setup
is called by source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh.

It has to be defined after source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh. Am I correct?

Reply:
Tbh, I don't think that e2e-tests.sh execute any codes, but only defines functions which can be called later. Otherwise `knative_setup` would have to be defined before sourcing this file

As a rule of thumb: Any shell script which is source must only define functions or global environment variables (if they have to).